### ISSUE TITLE
option -renames is confused by ' and &

### ISSUE THREAD

**Issue by felagund** [dave/gpt#3]
Run gpt -renames and then run it again on the new file, you will probably get this output:

```
"Villa O&amp;amp;apos;Higgins {36H, 36P, 37P, 37H}", "Villa O'Higgins {36H, 36P, 37P, 37H}"
"Punta Chana &amp;amp;amp; Desembocadura Rio Rayas", "Punta Chana & Desembocadura Rio Rayas"
"Lago O&amp;amp;apos;Higgins Brazo Nordeste", "Lago O'Higgins Brazo Nordeste"
"Lago O&amp;amp;apos;Higgins Brazo Desague", "Lago O'Higgins Brazo Desague"
"Lago O&amp;amp;apos;Higgins Brazo Poniente", "Lago O'Higgins Brazo Poniente"
"Cerro O&amp;amp;apos;Higgins", "Cerro O'Higgins"
"Lago O&amp;amp;apos;Higgins Brazo Sur", "Lago O'Higgins Brazo Sur"
"Lago O&amp;amp;apos;Higgins Brazo Sudoccidental", "Lago O'Higgins Brazo Sudoccidental"
"Glacier O&amp;amp;apos;Higgins", "Glacier O'Higgins"
"Lago O&amp;amp;apos;Higgins Brazo Maipu", "Lago O'Higgins Brazo Maipu"
"Settlement {04-02} [11.7/1032] (Private Resort &amp;amp;amp;#34;Sierra de Belavista&amp;amp;amp;#34;)", "Settlement {04-02} [11.7/1032] (Private Resort &#34;Sierra de Belavista&#34;)"
"Settler, Lodging $ {22N} [7.7/157] (Christian, &amp;amp;amp;#34;Las Nieves&amp;amp;amp;#34;)", "Settler, Lodging $ {22N} [7.7/157] (Christian, &#34;Las Nieves&#34;)"
"Settler, Lodging $ {22N} [17.4/158] (Soilo, &amp;amp;amp;#34;Castanos&amp;amp;amp;#34;)", "Settler, Lodging $ {22N} [17.4/158] (Soilo, &#34;Castanos&#34;)"
"Settler, Lodging $$ {22N} [28.9/53] (Ricardo Garado, &amp;amp;amp;#34;Cero Mesa&amp;amp;amp;#34;)", "Settler, Lodging $$ {22N} [28.9/53] (Ricardo Garado, &#34;Cero Mesa&#34;)"
"Settler {22N} [34.6/31] (Paulina &amp;amp;amp;amp; Jovino)", "Settler {22N} [34.6/31] (Paulina &amp; Jovino)"
"Settler {22N} [37.2+0.2/35] (Senora Oco, &amp;amp;amp;#34;La Junta&amp;amp;amp;#34;)", "Settler {22N} [37.2+0.2/35] (Senora Oco, &#34;La Junta&#34;)"
"Settler, Lodging $ {22-A} [1.0/38] (Nancy &amp;amp;amp;amp; Chindo, &amp;amp;amp;#34;Base Traidor&amp;amp;amp;#34;)", "Settler, Lodging $ {22-A} [1.0/38] (Nancy &amp; Chindo, &#34;Base Traidor&#34;)"
"Settler {22-06} [28.3+0.2/33] (Jaime, &amp;amp;amp;#34;Molino&amp;amp;amp;#34;)", "Settler {22-06} [28.3+0.2/33] (Jaime, &#34;Molino&#34;)"
"Lodging $$ {22-06B} [0.9+0.3/33] (&amp;amp;amp;#34;Isla Jabali&amp;amp;amp;#34;)", "Lodging $$ {22-06B} [0.9+0.3/33] (&#34;Isla Jabali&#34;)"
"Settler {33H} [76.7/535] (Maria &amp;amp;amp;amp; Julio)", "Settler {33H} [76.7/535] (Maria &amp; Julio)"
"Trailhead {37H-08} [65.3/354] (Difficult trail section &amp;amp;amp;#34;Paso Malo&amp;amp;amp;#34;)", "Trailhead {37H-08} [65.3/354] (Difficult trail section &#34;Paso Malo&#34;)"
"Trailhead {37H-08} [68.7/303] (Difficult trail section &amp;amp;amp;#34;Paso Malo&amp;amp;amp;#34;)", "Trailhead {37H-08} [68.7/303] (Difficult trail section &#34;Paso Malo&#34;)"
"Port {37P-09D} [15.0/267] (Puerto Refugio Glaciar O&amp;amp;apos;Higgins Bajo)", "Port {37P-09D} [15.0/267] (Puerto Refugio Glaciar O'Higgins Bajo)"
"Viewpoint {37P-13} [7.5/1057] (Glaciar O&amp;amp;apos;Higgins)", "Viewpoint {37P-13} [7.5/1057] (Glaciar O'Higgins)"
"Refuge {37P-14A} [3.7/284] (Refugio Glaciar O&amp;amp;apos;Higgins Bajo)", "Refuge {37P-14A} [3.7/284] (Refugio Glaciar O'Higgins Bajo)"
"Refuge {37P-15} [13.3/1219] (DGA Refugio Glaciar O&amp;amp;apos;Higgins Alto)", "Refuge {37P-15} [13.3/1219] (DGA Refugio Glaciar O'Higgins Alto)"
```

(for tracks, there are many more)

Probably something not being escaped in a print function, but I know zero Go.

**Comment by dave** [comment]
I can't see any bugs in the Go code... I can only imaging that those encoding problems were created by another process (e.g. not the gpt Go tool). I've added a bit of code that fixes all the encoding bugs for all the text strings in the file when it opens... this seems to fix it... Let me know if that works for you!

Released as v0.3.4

**Comment by felagund** [comment]
Haha, my report was indeed wrong. The tool in question that breaks is is none other than Google Earth. So running `gpt -renames` opening the new master file in Google Earth and saving it and running `gpt -renames` again in version 0.3.3 produced the output above. I can confirm that 0.3.4 fixes it. Thanks!

(I also run my connect_tracks from here: https://github.com/felagund/gpt_scripts but that was not the culprit :-))
